kotlin/workflow-core/src/main/java/com/squareup/workflow/WorkflowContext.kt

```diff
@@ -175,6 +179,59 @@ inline fun <StateT : Any, OutputT : Any, reified T> WorkflowContext<StateT, Outp
     handler
 )
 
+/**
+ * Subscribes to [flow] and invokes [handler] when it emits the next value.
+ *
+ * This allows us to use this method to, for example, monitor a stream of values emitted by a single
+ * fow over a series of state transitions. (Use [key] if you need to work simultaneously with
+ * several [Flow]s of the same type.)
+ * The subscription will be disposed when `compose` returns without passing a [Flow] of the same
+ * type and with the same key to this method.
+ *
+ * ## Experimental Status
+ *
+ * The [Flow] API is still marked experimental, and the compiler will warn you if you use it without
+ * opting in. The Channels API is also experimental, however Flow is much newer than channels,
+ * so this method is marked as experimental to discourage use in production code.
+ *
+ * @param key An optional string key that is used to distinguish between subscriptions of the same
+ * type.
+ */
+@FlowPreview
+inline fun <StateT : Any, OutputT : Any, reified T> WorkflowContext<StateT, OutputT>.onNext(
+  flow: Flow<T>,
+  key: String = "",
+  noinline handler: (ChannelUpdate<T>) -> WorkflowAction<StateT, OutputT>
+) = onNext(flow, KTypes.fromGenericType(Flow::class, T::class), key, handler)
+
+/**
+ * Subscribes to [flow] and invokes [handler] when it emits the next value.
+ *
+ * This allows us to use this method to, for example, monitor a stream of values emitted by a single
+ * fow over a series of state transitions. (Use [key] if you need to work simultaneously with
+ * several [Flow]s of the same type.)
+ * The subscription will be disposed when `compose` returns without passing a [Flow] of the same
```

Comment:
`compose` is already a stale name. I'd change it to a link s.t. it gets updated as renames continue.

Reply:
There's two compose methods though, and it would be weird to specifically link to one when they are the same concept with different shapes.